src/layer/vector/Canvas.js

```diff
@@ -253,7 +253,7 @@ L.Canvas = L.Renderer.extend({
 
 		for (var id in this._layers) {
 			layer = this._layers[id];
-			if (layer.options.interactive && layer._containsPoint(point)) {
+			if (layer.options.interactive && layer._containsPoint(point) && !this._map._draggableMoved(layer)) {
```

Comment:
Might it be possible to just remove the click listener while the map is dragged? I guess that would make it a bit clearer what's going on, and make it less coupled with `Map`'s internals.